### ISSUE TITLE
CI testing EBCDIC support

### ISSUE THREAD

**Issue by hlandau** [openssl/openssl#23496]
We don't seem to be CI testing our EBCDIC support. We probably should be.

**Comment by hlandau** [comment]
> Hmm... any Z/OS emulators around? 😁

There is Hercules. But you'd need to get the OS from IBM. Alternatively maybe someone could provide external CI?

**Comment by hlandau** [comment]
I'm being told that IBM Cloud z/OS is the "approved" way to get this sort of thing. As for the cost...

**Comment by hlandau** [comment]
@holger-dengler @ifranzki Any ideas?

**Comment by holger-dengler** [comment]
Although Ingo and I work in Linux development for System Z, I'll try to find out, if there is someone in z/OS development, which may help out here.

**Comment by hlandau** [comment]
Thanks!

**Comment by ifranzki** [comment]
Some time ago some EBCDIC related fixes where contributed by @opensslonzos-github ....

**Comment by opensslonzos-github** [comment]
It is unlikely that my employer would agree to this, but I will ask.
I think it would be far more likely to find a solution by asking IBM itself or a program they sponsor like some of the following.IBM Academic Initiative | Enterprise Systems | Walton College | University of Arkansas


| 
| 
| 
|  |  |

 |

 |
| 
|  | 
IBM Academic Initiative | Enterprise Systems | Walton College | Universi...

The University of Arkansas serves as an IBM hub for universities that are teaching mainframe and open source Lin...
 |

 |

 |




https://community.ibm.com/community/user/home

    On Wednesday, February 7, 2024 at 03:58:05 AM EST, Ingo Franzki ***@***.***> wrote:  
 
 


Some time ago some EBCDIC related fixes where contributed by @opensslonzos-github ....

**Comment by opensslonzos-github** [comment]
It would appear that someone at IBM is trying to make this sort of thing happen for open source projects like OpenSSL

https://tac.openmainframeproject.org/


AND 

Developer Resources for Linux on s390x - Open Mainframe Project


| 
| 
|  | 
Developer Resources for Linux on s390x - Open Mainframe Project

The zArchitecture (or s390x)-based processor in an IBM Z or LinuxONE is a big-endian computer architecture, and ...
 |

 |

 |
















    On Wednesday, February 7, 2024 at 12:55:31 PM EST, Sr Zos ***@***.***> wrote:  
 
  It is unlikely that my employer would agree to this, but I will ask.
I think it would be far more likely to find a solution by asking IBM itself or a program they sponsor like some of the following.IBM Academic Initiative | Enterprise Systems | Walton College | University of Arkansas


| 
| 
| 
|  |  |

 |

 |
| 
|  | 
IBM Academic Initiative | Enterprise Systems | Walton College | Universi...

The University of Arkansas serves as an IBM hub for universities that are teaching mainframe and open source Lin...
 |

 |

 |




https://community.ibm.com/community/user/home

    On Wednesday, February 7, 2024 at 03:58:05 AM EST, Ingo Franzki ***@***.***> wrote:  
 
 


Some time ago some EBCDIC related fixes where contributed by @opensslonzos-github ....

**Comment by hlandau** [comment]
The problem is that Linux on z will not be adequate for testing EBCDIC AFAIK.

**Comment by holger-dengler** [comment]
That is correct, Linux on IBM Z is using the same character sets like Linux on any other architecture.

**Comment by flabuc** [comment]
Hello,
I am currently one of the the developers at 21st Century Software (21CS), taking care of OpenSSL in VSEn.

Just as a background info, VSEn is the follow on of IBM's z/VSE. 

IBM announced in 2021 the end of support of z/VSE in Sep 2022 ,  and at same time 21CS acquired the rights to continue developing, improving and supporting VSEn for existing customers that will migrate to our version making the VSEn the natural choice of continuity for the OS and other related products.

VSEn (as well as z/VSE) is already using OpenSSL in EBCDIC environment, currently in V 1.1.1t+CVEs and we are working to make available the port to V 3.0.x LTS.

We do a series of internal tests covering several areas including the interaction with hardware specific cards. We are also interested  doing tests covering the EBCDIC specific features in OpenSSL, whenever something new or even fixes are created. Some tests are also carried by our Software Partners that produce additional utilities to run in VSEn

So, I believe we can strengthen our communication channels to do the continuity of what is needed for the continuity of EBCDIC support in OpenSSL.

**Comment by paulidale** [comment]
I am almost certain that EBCDIC support isn't up to scratch.
It really should be.

**Comment by opensslonzos-github** [comment]
Hello OpenSSL folks.
 Just wondering if there has been any movement on :
 a) fixing OpenSSL tests on EBCDIC systems? b) getting dev/test builds going at 21CS or any other entity willing to take that on.

Than you to folks at 21CS for their willingness to take on this effort!
- OpenSSLonzOS


    On Thursday, February 15, 2024 at 10:50:55 PM EST, Pauli ***@***.***> wrote:  
 
 


I am almost certain that EBCDIC support isn't up to scratch.
It really should be.

**Comment by t8m** [comment]
EBCDIC in CI is unlikely to ever happen. Which does not mean that having a community platform with EBCDIC is impossible, it would just mean that the community maintainer of such platform would have to do manual testing of EBCDIC support and provide patches if things break for them.
